### PR TITLE
LSIF: Target postgres for xrepo unit tests

### PR DESCRIPTION
**Pull request by efritz**
After we switched to postgres from sqlite, our tests remained in sqlite. This causes issues where we can't use postgres-specific features. This adds the machinery necessary to create test databases targeting a local postgres instance.

### CODE REVIEW

File: lsif/src/test-utils.ts [hunk 1]
```diff
@@ -2,36 +2,132 @@ import * as fs from 'mz/fs'
...
+    const env = `PGHOST=${host} PGPORT=${port} PGSSLMODE=disable`
+    const createCommand = `${env} createdb ${database}`
+    const dropCommand = `${env} dropdb --if-exists ${database}`
+    const migrateCommand = `${env} PGDATABASE=${database} ${migrateScriptPath} up`
+
+    // Create cleanup function to run after test
+    const cleanup = (): Promise<void> => child_process.exec(dropCommand).then(() => {})
+
+    // Try to create database
+    await child_process.exec(createCommand)
```
Comment (chrismwendt) [inline lsif/src/test-utils.ts]: I think `exec` takes a second options argument where you can pass `env: { ...process.env, PGHOST: '...' }`. See https://stackoverflow.com/questions/7824789/nodejs-set-environment-variable-for-exec#comment57015051_7825732

File: lsif/.gitignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Run/test artifacts
+lsif-storage
+cpp-*
+typescript-*
+xrepo-*
```
Comment (chrismwendt) [inline lsif/.gitignore]: Do these get created at the top-level? It'd be nice to keep the top level minimal. I see there's already a `test-data/` directory specified.

File: lsif/src/xrepo.models.ts [hunk 3]
```diff
@@ -147,49 +147,12 @@ export class ReferenceModel extends Package {
...
-
-    @ViewColumn()
-    public commit!: string
-
-    @ViewColumn()
-    public parentCommit!: string
-
-    @ViewColumn()
-    public hasLsifData!: boolean
-}
```
Comment (chrismwendt) [inline lsif/src/xrepo.models.ts]: This was unused, right?

File: lsif/.gitignore (hunk 2 above)
Comment (efritz) [reply lsif/.gitignore]: This is for temp files during test that weren't getting cleaned up - I just now figured out why. Will remove.

File: lsif/src/xrepo.models.ts (hunk 3 above)
Comment (efritz) [reply lsif/src/xrepo.models.ts]: It's used but only textually in queries now. No reason to have it as it won't be created by typeorm.

File: dev/drop-test-databases.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
```
Comment (beyang) [inline dev/drop-test-databases.sh]: @efritz can you CODEOWNERS this file to you and Chris?

Comment (efritz) [reply dev/drop-test-databases.sh]: Will do.

**Pull request merged**